src/payment/data/sagas.test.js

```diff
@@ -148,12 +164,73 @@ describe('saga tests', () => {
         basketDataReceived(transformResults(response.data)),
         clearMessages(),
         basketDataReceived(transformResults(response.data)),
+        basketProcessing(false),
+        fetchBasket.fulfill(),
+      ]);
+      expect(caughtErrors).toEqual([]);
+      expect(mockApiClient.get).toHaveBeenCalledTimes(3);
+      expect(mockApiClient.get).toHaveBeenCalledWith(`${configuration.ECOMMERCE_BASE_URL}/bff/payment/v0/payment/`);
+      expect(mockApiClient.get).toHaveBeenCalledWith(
+        `${configuration.LMS_BASE_URL}/api/discounts/course/${courseKey}`,
+        {
+          xhrFields: { withCredentials: true },
+        },
+      );
+    });
+
+    it('should update basket data with jwt on second call', async () => {
```

Comment:
Added this test to prove that we use the discount_jwt parameter on the second getBasket call if it was provided by the discount check.